etc/pr_check/match.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/google/go-github/v54/github"
+)
+
+// prCheck indicates which of the implemented checks 'actOnMatchingPRs' will
+// apply to scanned PRs.
+type prCheck byte
+
+const (
+	missingJiraTicket prCheck = iota
+	createdLastMinute
+)
+
+func (c prCheck) String() string {
+	switch c {
+	case missingJiraTicket:
+		return "PR contains Jira ticket"
+	case createdLastMinute:
+		return "PR reviewed before release"
+	}
+	return "Error: unrecognized pr check"
+}
+
+// prAction indicates which of the implemented actions 'actOnMatchingPRs' will
+// apply to scanned PRs.
+type prAction byte
+
+const (
+	commentOnPr prAction = iota
+	addStatus
+)
+
+// jiraRE is the regex that a PR's title+body+HEAD branch must match to pass the
+// 'missingJiraTicket' check.
+var jiraRE = regexp.MustCompile(`\[[A-Z]{2,4}-[0-9]{1,5}\]`)
```

Comment:
I don't think Jira requires that the project be capitalized (it understands a branch like core-1234-foo-bar).